src/NuGet.Core/NuGet.DependencyResolver.Core/Remote/RemoteWalkContext.cs

```diff
@@ -26,7 +22,8 @@ public RemoteWalkContext(SourceCacheContext cacheContext, PackageSourceMapping p
             RemoteLibraryProviders = new List<IRemoteDependencyProvider>();
             PackageSourceMapping = packageSourceMapping ?? throw new ArgumentNullException(nameof(packageSourceMapping));
 
-            FindLibraryEntryCache = new ConcurrentDictionary<LibraryRangeCacheKey, Task<GraphItem<RemoteResolveResult>>>();
+            FindLibraryEntryCache = new TaskResultCache<LibraryRangeCacheKey, GraphItem<RemoteResolveResult>>();
+            ResolvePackageLibraryMatchCache = new TaskResultCache<LibraryRange, Tuple<LibraryRange, RemoteMatch>>();
```

Comment:
This will probably be empty most of the time. 
That's why I originally had the cache outside of the remote walk context. 
It's not doing so much walking, since PackageDownload is kind of an extra hook to restore, so logically I had kept it out of the regular restore flow on purpose. 

You don't have to change this, I'm just explaining the reasoning, as their separation was intentional.

Reply:
I mainly liked this change because I didn't think the caches should be `public` since they're only used as an implementation detail and callers don't need to concern themselves with that.  Then since this Task cache was here, I figured I'd put the other one here as well.  Just my throught process.

Reply:
Yeah, I'm fine with that. 

When I get around to do: https://github.com/NuGet/Home/issues/13077, it'll make a bunch of our restore changes straightforward.